File: Java-Solutions/Problem4.java
// Problem 4 - Largest Palindrome Product

public class Problem4 {

    static boolean palindromeCheck(int number) {
        String rawString = Integer.toString(number);
        String revString = "";

        for (int i=rawString.length()-1; i>=0; i--) {
            revString+=rawString.charAt(i);
        }

        // System.out.println("Og string: " + rawString);
        // System.out.println("Rev string: " + revString);
        // System.out.println();

        if (rawString.equalsIgnoreCase(revString)){
            return true;
        }
        return false;
    }

    public static void main(String[] args) {
        int largestPalindrome = 0;

        for (int i=100;i<1000;i++) {
            for (int j=100; j<1000; j++) {
                if (palindromeCheck(i*j)==true && largestPalindrome<i*j) {
                    largestPalindrome = i*j;
                    // System.out.printf("Current largest: %d", largestPalindrome);
                    // System.out.println();
                }
            }
        }

        System.out.printf("The largest palindrome product is: %d", largestPalindrome);
        System.out.println();


    }

}

File: Python-Solutions/Problem4.py
#Problem 4 - Largest Palindrome Product

def palindromeCheck(number):
    if str(number)[::-1].lower() == str(number).lower():
        return True
    return False 


largestPalindrome = 0
for i in range(100, 999):
    for j in range(100, 999):
        if palindromeCheck(i*j) == True:
            if largestPalindrome<i*j:
                largestPalindrome = i*j

print(f"Largest palindrome product made by 2 three-digit numbers is: {largestPalindrome}")